roles/network_plugin/calico/templates/calico-node.yml.j2

```diff
@@ -53,6 +56,11 @@ spec:
                 configMapKeyRef:
                   name: calico-config
                   key: cluster_type
+            # Set noderef for node controller.
```

Comment:
Any more details around this parameter?  We specify NODENAME below that uses the same `spec.nodeName`

Reply:
I was going to link you to the Calico docs for it but I see that it is missing.  That is used for keeping track of what K8s's name for the host the pod is running on. The main use case is for the calico-kube-controller which can monitor K8s for node deletions/removals and clean up the pods and info associated with that host.  This isn't really necessary right now for kubespray as the configuration for the calico 'policy-controller' is the default which does not include the node controller. (`ENABLED_CONTROLLERS` in [reference docs](https://docs.projectcalico.org/v2.6/reference/kube-controllers/configuration), see the defaults [here](https://github.com/projectcalico/kube-controllers/blob/master/pkg/config/config.go#L32) )